pkg/api/v1/conversion.go

```diff
@@ -480,10 +480,9 @@ func Convert_v1_PodTemplateSpec_To_api_PodTemplateSpec(in *PodTemplateSpec, out
 		in.Spec.InitContainers = values
 
 		// Call defaulters explicitly until annotations are removed
-		for i := range in.Spec.InitContainers {
-			c := &in.Spec.InitContainers[i]
-			SetDefaults_Container(c)
-		}
+		tmpPod := &Pod{Spec: PodSpec{InitContainers: values}}
+		SetObjectDefaults_Pod(tmpPod)
```

Comment:
hmm, this won't catch all cases... things like this propagate other parts of the pod spec down into the containers:
```
	if obj.HostNetwork {
		defaultHostNetworkPorts(&obj.Containers)
		defaultHostNetworkPorts(&obj.InitContainers)
	}
```